### PR TITLE
Continue training from a PLY file

### PR DESCRIPTION
**Pull request by NotCompsky**
Resolves [feature request](https://github.com/pierotofy/OpenSplat/issues/83).

* Adds `--load-snapshot` option, to specify the path of the `PLY` file to load
* Adds iteration count to the `comment` in the `PLY` outputs
* Verifies `PLY` inputs adhere to `OpenSplat`'s output format
  * It won't load `PLY` files that people have edited, i.e. where they've changed the order of header items, or where they've added more comments into the header - that would double the complexity for almost no gain

It works for me using CPU. I believe it should work on GPU too - if `tensor = tensorCpu.detach().to(device).requires_grad_();` is the correct way to copy the data to the processing device - but it's been a while since I've used GPU computing.

***

For those of you who have existing `.ply` files that you want to continue training from, when this PR is merged, this is how you'd do it:

Edit the `.ply` files, to change this line:

    comment Generated by opensplat

to this:

    comment Generated by opensplat at iteration 12345

where `12345` is whatever the iteration number is that you want to continue training from.

### CODE REVIEW

**Comment by pierotofy** [comment]
This looks great! I've fixed a memory leak in the optimizer initialization and renamed the parameter to `--resume` (also some minor refactoring), tested on GPU and seems to work well!

Thanks for the PR.
